src/ui/public/errors.js

```diff
@@ -173,53 +171,51 @@ _.class(errors.DuplicateField).inherits(KbnError);
 errors.SavedObjectNotFound = function SavedObjectNotFound(type, id) {
   this.savedObjectType = type;
   this.savedObjectId = id;
-  const idMsg = id ? ' (id: ' + id + ')' : '';
+  const idMsg = id ? ` (id: ${id})` : '';
   KbnError.call(this,
-    'Could not locate that ' + type + idMsg,
+    `Could not locate that ${type}${idMsg}`,
     errors.SavedObjectNotFound);
 };
 _.class(errors.SavedObjectNotFound).inherits(KbnError);
 
 /**
  * Tried to call a method that relies on SearchSource having an indexPattern assigned
  */
-errors.IndexPatternMissingIndices = function IndexPatternMissingIndices(type) {
+errors.IndexPatternMissingIndices = function IndexPatternMissingIndices(message) {
+  const defaultMessage = 'IndexPattern\'s configured pattern does not match any indices';
   KbnError.call(this,
-    'IndexPattern\'s configured pattern does not match any indices',
+    message.length ? `No matching indices found: ${message}` : defaultMessage,
```

Comment:
checking message that message is "truthy" is probably a little bit safer long term.